.github/workflows/presubmit.yml

```diff
@@ -94,35 +103,26 @@ jobs:
           set -x
           AFFECTED_FILES=`echo "${{ steps.changed-files.outputs.files_including_removals }}" | sed 's|\([^ ]\+\)|--changedFilePath=\1|g'`
           echo "::set-output name=files::$AFFECTED_FILES"
-      - name: "warm up gradle"
-        id: warm-up-gradle-cache
-        uses: gradle/gradle-command-action@v1
-        env:
-          JAVA_HOME: ${{ steps.setup-java.outputs.path }}
-          JAVA_TOOLS_JAR: ${{ steps.setup-tools-jar.outputs.toolsJar }}
+
+      - name: "Setup Gradle"
+        uses: gradle/gradle-build-action@v2
         with:
-          arguments: tasks -PandroidXUnusedParameter=activity # add project name to cache key
-          build-root-directory: activity
-          configuration-cache-enabled: false
-          dependencies-cache-enabled: true
-          dependencies-cache-key: |
-            **/libs.versions.toml
-          dependencies-cache-exact: false
-          gradle-executable: activity/gradlew
-          wrapper-directory: activity/gradle/wrapper
-          distributions-cache-enabled: true
+          # Only save Gradle User Home state for builds on the 'androidx-main' branch.
+          # Builds on other branches will only read existing entries from the cache.
+          cache-read-only: ${{ github.ref != 'refs/heads/androidx-main' }}
+          # Don't share cache entries with other jobs
+          gradle-home-cache-strict-match: true
+          # Limit the size of the cache entry
+          gradle-home-cache-excludes: | 
+            caches/jars-9
```

Comment:
Please add comments on what is in these directories so we know what we are dropping. Ideally link to Gradle docs.

Reply:
Done. Unfortunately these directories are undocumented.